### ISSUE TITLE
vmagent scrape targets prometheus federate, metrics  up disappeared

### ISSUE THREAD

**Issue by Yifengzis** [VictoriaMetrics/VictoriaMetrics#3557]
# ./vmagent-prod -version
vmagent-20221220-225543-tags-v1.85.3-0-gf443fad56

vmagent scrape targets prometheus federate, metrics  up disappeared
for example: 
up{instance="localhost:9090", job="prometheus"}

However, it is normal to write remotely on the Prometheus Agent. [prometheus-2.41.0]

![image](https://user-images.githubusercontent.com/43021839/209825949-10a7df26-75de-4778-a63a-ae3aa8449577.png)

**Comment by dmitryk-dk** [comment]
Hi @Yifengzis ! Can you check it on `graph` tab and enable `Disable cache`?

**Comment by Yifengzis** [comment]
> Hi @Yifengzis ! Can you check it on `graph` tab and enable `Disable cache`?
vmagent scrape targets prometheus federate, metrics "up" disappeared
![image](https://user-images.githubusercontent.com/43021839/209912429-b7fb649a-ad2e-4cce-9e5d-d3d5ebd5d2ed.png)

**Comment by dmitryk-dk** [comment]
Hi @Yifengzis ! So when you disable cache you metrics become visible?

**Comment by Yifengzis** [comment]
> metrics

Metrics  is not visible.

**Comment by dmitryk-dk** [comment]
Can you check logs of the vmalert and share them if there is something wroong? Also can you export data with `api/v/1/export` with that is more than 30 minutes?

**Comment by Yifengzis** [comment]
> Can you check logs of the vmalert and share them if there is something wroong? Also can you export data with `api/v/1/export` with that is more than 30 minutes?

**I'm sorry, my English is not very good!**

####  "VMagent" remote write has the following indicators less than "PrometheusAgent"

up{instance="localhost:9100",job="node_exporter"} 0
up{instance="prometheus-server:19090",job="prometheus"} 1

#### See the following figure:
![image](https://user-images.githubusercontent.com/43021839/209959725-73ac9fdd-b56b-4634-ac21-60696ec527da.png)

**Comment by Yifengzis** [comment]
@dmitryk-dk 

The exported data and test cases have been emailed to you. Please check

#### vmagent/config/prometheus.yml 

```yaml
$ cat vmagent/config/prometheus.yml 
global:
  scrape_interval: 15s
  external_labels:
    source: vmagent

scrape_configs:
  - job_name: "federate"
    scrape_interval: 15s
    honor_labels: true
    metrics_path: "/federate"
    params:
      match[]:
        - '{__name__!=""}'
    static_configs:
      - targets:
          - prometheus-server:19090
```

#### prometheus-agent/config/prometheus.yml

```yam
$ cat prometheus-agent/config/prometheus.yml 
global:
  scrape_interval: 15s
  evaluation_interval: 15s
  external_labels:
    source: prometheus-agent

scrape_configs:
  - job_name: "federate"
    scrape_interval: 15s
    honor_labels: true
    metrics_path: "/federate"
    params:
      match[]:
        - '{__name__!=""}'
    static_configs:
      - targets:
          - prometheus-server:19090

remote_write:
  - url: http://victoriametrics:8428/api/v1/write
```

#### prometheus-server/config/prometheus.yml

```yaml
global:
  scrape_interval: 15s
  evaluation_interval: 15s

scrape_configs:
  - job_name: "prometheus"
    static_configs:
      - targets: 
        - prometheus-server:19090

  - job_name: "node_exporter"
    static_configs:
      - targets: 
        - localhost:9100
```

#### test_docker-compose.yml

```yaml
version: '3.9'

services: 
  victoriametrics: 
    container_name: victoriametrics
    image: victoriametrics/victoria-metrics:v1.85.3
    restart: always
    hostname: victoriametrics
    command:
      - "--storageDataPath=/storage"
      - "--httpListenAddr=:8428"
    volumes:
      - /etc/localtime:/etc/localtime:ro
      #- ./victoriametrics/data:/storage
    ports:
      - 8428:8428
    networks:
      - vm_net

  vmagent:
    container_name: vmagent
    image: victoriametrics/vmagent:v1.85.3
    restart: always
    hostname: vmagent
    command:
      - "--httpListenAddr=:8429"
      - "--promscrape.config=/etc/prometheus/prometheus.yml"
      - "--remoteWrite.url=http://victoriametrics:8428/api/v1/write"
    depends_on:
      - "victoriametrics"
    volumes:
      #- ./vmagent/data:/vmagent-remotewrite-data
      - ./vmagent/config:/etc/prometheus
    ports:
      - 8429:8429
    networks:
      - vm_net

  prometheus-agent:
    container_name: prometheus-agent
    image: prom/prometheus:v2.41.0
    restart: always
    hostname: prometheus-agent
    command:
      - --web.listen-address=:9090
      - --config.file=/prometheus/config/prometheus.yml
      - --enable-feature=agent
      - --storage.agent.path=/prometheus/data-agent
    volumes:
      - /etc/localtime:/etc/localtime:ro
      #- ./prometheus-agent/data:/prometheus/data-agent
      - ./prometheus-agent/config:/prometheus/config
    ports:
      - 9090:9090
    networks:
      - vm_net

  prometheus-server: 
    container_name: prometheus-server
    image: prom/prometheus:v2.41.0
    restart: always
    hostname: prometheus-server
    command:
      - --web.listen-address=:19090
      - --config.file=/prometheus/config/prometheus.yml
      - --storage.tsdb.path=/prometheus/data
    volumes:
      - /etc/localtime:/etc/localtime:ro
      #- ./prometheus-server/data:/prometheus/data
      - ./prometheus-server/config:/prometheus/config
    ports:
      - 19090:19090
    networks:
      - vm_net

networks:
  vm_net:
```

**Comment by dmitryk-dk** [comment]
Hi! Thank you for your information i will investigate it

**Comment by dmitryk-dk** [comment]
Hi @Yifengzis ! I have imported your data to the VM and I can see your data when using the query
<img width="1680" alt="Screenshot 2023-01-03 at 12 32 20" src="https://user-images.githubusercontent.com/12189570/210340465-f33a5011-b43c-42de-ac05-2d89373afc85.png">

Query sends the request to `api/v1/query_range` and this API returns also ephemeral points to show your request on the graph.
Please check this part of the documentation [query_range](https://docs.victoriametrics.com/keyConcepts.html#range-query)

When you are trying to check the table or JSON tab it sends an instant query to `api/v1/query` this API returns only points that can be found by provided step. Please check this part of the documentation [instant](https://docs.victoriametrics.com/keyConcepts.html#instant-query)
When you decrease your start point you will see data in the table.
<img width="1413" alt="Screenshot 2023-01-03 at 12 42 18" src="https://user-images.githubusercontent.com/12189570/210341843-3873e411-7857-45ba-a79f-dd72681a6d4b.png">

**Comment by Yifengzis** [comment]
Hi, @dmitryk-dk ! 

I think you may have misunderstood me. What I want to say is that it is normal to write to "VM" remotely through "Prometheus", but the "UP" indicator will be missing when writing to "VM" remotely through "VMAgent"

**Comment by capull0** [comment]
We have the same problem. We migrated our Prometheus to Victoria Metrics.
We are federating several Prometheus instances from big Kubernetes clusters with vmagent.
The "up"-metric is missing. The same scrape config is working with the old Prometheus.

stream parsing mode should be enabled, because we are scraping a huge amount of metrics.

We are using version: 1.85.0

**Comment by dmitryk-dk** [comment]
Hi @Yifengzis ! Hi @capull0 ! Can you please check this request
```curl http://<victoriametrics-addr>:8428/federate -d 'match[]=<timeseries_selector_for_export>' -d 'start=1654543486' -d 'end=1654543486'```
Like it described in the documentation [federation](https://docs.victoriametrics.com/?highlight=federate#federation)

Please set your `match` param correctly and `start, end` params.

Does it return any value?

**Comment by dmitryk-dk** [comment]
Tomorrow I will check the setup which @Yifengzis sent before!

**Comment by capull0** [comment]
I think we found the problem.
Victoria Metrics shows different behavior than Prometheus.
Even you configure honor_labels to true, VM adds "exported_" prefix to the up metric, means "up"-metric is not missing, it's renamed to exported_up.
It's documented here: https://docs.victoriametrics.com/vmagent.html#automatically-generated-metric , so seems like it's "by design".
But this should not be the expected behavior.

**Comment by dmitryk-dk** [comment]
Hi @capull0 and @Yifengzis ! I have tested your setup. Possibly it is a bug. We will fix it soon.
Thank you.

**Comment by dmitryk-dk** [comment]
Hi @capull0 , @Yifengzis ! 
What we have now. After my tests I can say that we added this functionality in the commit [https://github.com/VictoriaMetrics/VictoriaMetrics/commit/654e94f420c071b4d258b7496b7b6d6a33516e98](https://github.com/VictoriaMetrics/VictoriaMetrics/commit/654e94f420c071b4d258b7496b7b6d6a33516e98)

And at this moment it works as described below: 

1. VictoriaMetrics stack
If we enable `honor_labels`, we got
```
up{instance="prometheus-server:19090",job="federate",source="vmagent"}

exported_up{instance="prometheus-server:19090",job="prometheus",source="vmagent"}
```
<img width="1680" alt="Screenshot 2023-01-24 at 23 46 18" src="https://user-images.githubusercontent.com/12189570/214530331-0b17726f-4875-4278-b843-3ebf2e2e61e5.png">


If we disable  `honor_labels`, we got:
```
up{instance="prometheus-server:19090",job="federate",source="vmagent"}

exported_up{exported_instance="prometheus-server:19090",exported_job="prometheus",instance="prometheus-server:19090",job="federate",source="vmagent"}
```
<img width="1680" alt="Screenshot 2023-01-24 at 23 51 03" src="https://user-images.githubusercontent.com/12189570/214530393-511df25d-1ae9-42bb-a301-df2be508ba7e.png">

Prometheus stack:
If we enable `honor_labels`, we got:
```
up{instance="prometheus-server:19090",job="federate",source="prometheus-agent"}

up{instance="prometheus-server:19090",job="prometheus",source="prometheus-agent"}
```
<img width="1678" alt="Screenshot 2023-01-24 at 23 59 39" src="https://user-images.githubusercontent.com/12189570/214531117-1e97fee3-11fe-4ce7-b175-b49f37ec1d47.png">



If we disable `honor_labels`,  we got:
```
up{exported_instance="prometheus-server:19090", exported_job="prometheus",instance="prometheus-server:19090",job="federate",source="prometheus-agent"}

up{instance="prometheus-server:19090"job="federate"source="prometheus-agent"}
```
<img width="1680" alt="Screenshot 2023-01-25 at 00 02 40" src="https://user-images.githubusercontent.com/12189570/214530543-44488a27-0ba5-449e-9dde-f6cef4f14c5b.png">

Prometheus didn't set  `exported_` to metric names, it is the difference

The explanation of this changes you can find here
[https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3406#issuecomment-1331288525](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3406#issuecomment-1331288525)
for the issue [https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3406](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3406)

@valyala , @hagen1778 what do you think should we change the behavior to be compatible with Prometheus or we will stay with our realization?

**Comment by capull0** [comment]
I think the easiest way should be: make the behavior configurable?

**Comment by valyala** [comment]
We decided implementing the following fix: if `honor_labels` option is explicitly set to `true` at the [scrape_config](https://docs.victoriametrics.com/sd_configs.html#scrape_configs), then do not add `exported_` prefix to metric names, which clash with the [automatically generated metric names](https://docs.victoriametrics.com/vmagent.html#automatically-generated-metrics). This fix has been included in [v1.87.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing this issue as fixed then.

@capull0 , feel free re-opening this issue or creating new one linked to this one if `vmagent v1.87.0` still doesn't work as expected in your case.
